### DISCUSSION TITLE
Updating Parent Timestamps when using Pivot Table

### DISCUSSION THREAD

**Discussion by intrepidws** [filamentphp/filament#6177] [Help] [answered]
I have a `Registrant` and a `Code` model, with a many to many relationship between them. I have a resource setup for each model as well as a relation manager for the other half of the relationship.

In `CodesRelationManager` for the `RegistrantResource`, I am able to use the following code to ensure that the `Registrant` timestamps get touched when a `Code` is attached to that `Registrant`:

```php
  protected function afterAttach(): void
  {
      $this->ownerRecord->touch();
  }
```

However, I'm not sure what to do when a `Registrant` is attached to a `Code` through the `RegistrantsRelationManager` on the `CodeResource`. I can't use the same `afterAttach` lifecycle hook, because I don't want to update the timestamp of the `Code` model - I want to update the timestamp of the `Registrant` that was just attached to the `Code`.

**Comment by danharrin** [comment]
Maybe https://laravel.com/docs/10.x/eloquent-relationships#touching-parent-timestamps

**Reply by intrepidws** [reply]
Right, that's what I looked at first but it won't work with a pivot table unfortunately.

Is there some way to access the record of the model being attached, from the relation manager code?

**Reply by danharrin** [reply, marked as answer]
In the callback on the action, you could try `->after(fn ($data) => $data['recordId'])` to get the ID

**Reply by intrepidws** [reply]
Thanks, that works great for the attach action. It doesn't work for the detach action though (I also tried `before`) - any ideas for that?

**Reply by danharrin** [reply]
I would imagine that detach has `$record` available

**Reply by intrepidws** [reply]
Thanks Dan.
